### DISCUSSION TITLE
Beta planning

### DISCUSSION THREAD

**Discussion by IHateMyKite** [IHateMyKite/UnforgivingDevices#124] [Polls]
There were less bug reports then I expected. I don't know if its either because not much people read support thread, or there are just not bugs. Either way, I thing it will be good idea next time to release the beta on the main page (as I was doing it before). This should hopefully help to gather much more feedback. But it will also obviously cause confusion, as many people will not try it if they will not be able to directly upgrade to main release from beta. So, should I upload it on main page, or do it as before ?

**Comment by kurotatsu77** [comment]
I think better to post it at the thread with clear statement that target DD version is 5.2 Beta 11. Those who don't read threads will not DL DD beta and thus are useless, so posting it as download with produce spike of false reports because of people trying to run it on DD 5.1.

**Comment by IHateMyKite** [comment]
But UD should work with 5.1 (at least the main mod, but I didn't test it maybe I'm wrong). Only issue are patches. If user uses 5.1 patch, they should have no issue

**Reply by kurotatsu77** [reply]
You're right. But I really think we should put the concentrated effort to make sure DD5.2 is fully supported by UD at it's release. If we spread effort to 5.1 and 5.2 it might be worse?

**Reply by IHateMyKite** [reply]
That is true, but we also need to get more bug reports, so we can fix it. Otherwise we will publish 2.1 release, and find out that half of the features don't work, because of some errors or compatibility. 5.1 is now fully supported, and as the 5.2 is directly backward comaptbile with 5.1, there should be no issue with 5.2. Only patches are what needs to be kept updated with 5.2. Only big change with 5.2 is the expression system, but as the system is directly taken from UD, I see no issue.

**Comment by iiw2012** [comment]
Posting to the main download section is fine as long as everyone understands they are downloading a beta. As long as the list of files will have a stable version on a par with the beta, I do not see any problems for users.
Perhaps even one of the new users will write in the support thread.

**Comment by IHateMyKite** [comment]
As most features are now ready, I will try to release the beta as soon as remaining PR are merged. I will release on download page (as it was done with 2.0 betas), so there are more potential testers. If anyone still ones to implement some changes, then do it quick or wait until beta is over. Once beta is released, I will only focus on bug fixes and small adjustments. **No big changes will be added, and I will not merge them until beta is over.**

**Reply by iiw2012** [reply]
I'm currently working on new custom animations. I have no plans to add new features.

**Reply by IHateMyKite** [reply]
Do you intend to release the animations later on LL as new mod or will you keep it on GitHub ? So I know how should I point in description later

**Reply by iiw2012** [reply]
Initially, I planned that animations would be part of the UD. The fact that they ended up in an independent entity is a nice bonus.

I would prefer that they be distributed with UD. Either as a simple archive or as an option in the UD installer. But this is purely your decision.

If you consider this option not suitable, then I will post the animations as a separate mod on LL.

Probably there are other options, ready for any.

**Reply by IHateMyKite** [reply]
I like the idea with it being part of fomod, but issue is that every time there will be update to animations, the whole mod will need to be re-uploaded. I thing the best option is if you upload them as new mod in LL and I just add link to them in mod description and in fomod. This way you can work on animations indecently from main mod.

**Reply by iiw2012** [reply]
I've published animations on LL. Please tell me if I missed something or made a mistake somewhere.
https://www.loverslab.com/files/file/26880-unforgiving-devices-extra-animations/

**Comment by kurotatsu77** [comment]
I'm good with feature freeze, got it.

**Comment by IHateMyKite** [comment]
Anyway, now that everything is implemented and the last changes from [Ethryan](https://github.com/Ethryan) integrated, I will start to writte up the changelog and release the beta (hopefully today)

**Comment by IHateMyKite** [comment]
I'm going to release the next and hopefully last beta today. It should hopefully fix most issues some users were having. There is only last thing I want to do before the release, and that is to complete the documentation. As there are a bunch of modules, I would ask you guys for help. 

@iiw2012, can you please document the modules UD_AnimationManager and UD_WidgetControl? You were the one working on it the most, so I think you are most suited for it. I see you already did some inter documentation with comments, so converting it to Natural Docs format should be no issue. You can check UD_API for some simple examples on how the syntax work.

@kurotatsu77, @Ethryan  I would only ask you if you could please check grammar of finished documentation. I was writing it fastly, so there is most likely a bunch of errors. The most recent documentation is always present in \docs\Documentation folder(only in NB or betas, it's not under source control). If you find an issue, you can just edit the comment in the script, or report where is it wrong and I will fix it.

I will focus on documenting UD_CustomDevice_RenderScript which will probably be biggest part of documentation yet.

In case you do some changes to documentation, you can test the changes by regenerating it by running batch script BUILDDOCU.bat which is in docs folder

**Reply by kurotatsu77** [reply]
Got it, will go through it.

**Reply by Ethryan** [reply]
Got it

**Reply by iiw2012** [reply]
Ok

**Comment by IHateMyKite** [comment]
**Quick update**, once the [PR155 ](https://github.com/IHateMyKite/UnforgivingDevices/pull/155) is reviewed and merged, I will release next, and hopefully last beta. This time, it should only work as a check, that implemented fixes are working. If they are, I will release main 2.1 shortly after. I would like to release the beta either today or tomorrow, so please, anyone who gets the time, give the PR look and do a review. I have tested it quite thoroughly, so only checking if nothing is fundamentally broken should be enough.

**Reply by kurotatsu77** [reply]
Will do.

**Comment by IHateMyKite** [comment]
Beta 5 is released. This should (yet again) be last beta. If not some additional game breaking bug appears, I will upload the 2.1 in mid of june. If bug appears -> beta 6. Hopefully it will not end like with release of 2.0 which needed 15 betas...

**Comment by IHateMyKite** [comment]
Update. I'm little concerned on how to continue. I wanted to fix the MCM issue some users had reported, but didn't get any info which could help to solve the problem.  But I didn't get any answers in long time. So instead of waiting, I will hope that releasing next beta will let me know if some users still have issues. I will also update the poll so users who are dont want to comment can just use poll. Anyway, I will update beta 6 today as there is basically nothing else to do.

**Reply by kurotatsu77** [reply]
OK, will publish US updated with scripts compiled to it shortly.
